qa/zcash/create_benchmark_archive.py

```diff
@@ -0,0 +1,236 @@
+import binascii
+import json
+import plyvel
+import progressbar
+import os
+import struct
+import subprocess
+import sys
+
+ZCASH_CLI = './src/zcash-cli'
+USAGE = """
+Requirements:
+- faketime
+- tar
+- %s (edit ZCASH_CLI in this script to alter the path)
+- A running mainnet zcashd using the default datadir with -txindex=1
+
+Example usage:
+
+make -C src/leveldb/
+virtualenv venv
+. venv/bin/activate
+pip install --global-option=build_ext --global-option="-L$(pwd)/src/leveldb/" --global-option="-I$(pwd)/src/leveldb/include/" plyvel
+pip install progressbar2
+LD_LIBRARY_PATH=src/leveldb python qa/zcash/create_benchmark_archive.py
+""" % ZCASH_CLI
+
+def check_deps():
+    if subprocess.call(['which', 'faketime', 'tar', ZCASH_CLI], stdout=subprocess.PIPE):
+        print USAGE
+        sys.exit()
+
+def encode_varint(n):
+    v = bytearray()
+    l = 0
+    while True:
+        v.append((n & 0x7F) | (0x80 if l else 0x00))
+        if (n <= 0x7F):
+            break
+        n = (n >> 7) - 1
+        l += 1
+    return bytes(v)[::-1]
+
+def decode_varint(v):
+    n = 0
+    for ch in range(len(v)):
+        n = (n << 7) | (ord(v[ch]) & 0x7F)
+        if (ord(v[ch]) & 0x80):
+            n += 1
+        else:
+            return n
```

Comment:
I confirmed that this matches https://github.com/zcash/zcash/blob/master/src/serialize.h#L309L375 .